root/frontend/src/App.jsx

```diff
@@ -55,32 +55,137 @@ class App extends Component {
+      return response.json();
+    }).then( (res) => {
+      //localStorage.setItem('username', data.username);
+      //localStorage.setItem('token', res.token);
+      console.log("got here, ", res);
+      //dispatch(userLoggedIn(data.username));
+    }).catch( (e) => console.error(e) );
+
   }
 
   /*** when user submits new information for a custom class (called by AddCustomClass) ***/
   onAddClassSubmit = (newClassObj, status) => {
     // fix name format by just taking class category and number
-    let nameParts = newClassObj.Id.match(/([A-Z]{4})(.*)([\d]{4})/);
-    newClassObj.Id = nameParts[1] + ' ' + nameParts[3];
+    let nameParts = newClassObj.id.match(/([A-Z]{4})(.*)([\d]{4})/);
+    newClassObj.id = nameParts[1] + ' ' + nameParts[3];
     // add / remove class id to the correct list
     if (status === 'Taken') {
-      this.markClassTaken(newClassObj.Id);
+      this.markClassTaken(newClassObj.id);
     } else {
-      this.markClassPlanned(newClassObj.Id);
+      this.markClassPlanned(newClassObj.id);
     }
 
     // if is already in flowchart, alert the user, but still add it to the list
-    if (newClassObj.Id in this.state.ClassDesc) {
+    if (newClassObj.id in this.state.ClassDesc) {
       this.setState({ showAlert: ['danger', `Class already exists! It was added to your flowchart as a ${status} class.`] });
       return;
     }
 
     // modify objects to include new class
-    let newClassDesc = { ...this.state.ClassDesc, [newClassObj.Id]: newClassObj };
+    let newClassDesc = { ...this.state.ClassDesc, [newClassObj.id]: newClassObj };
     this.setState({
       ClassDesc: newClassDesc,
-      AddedClasses: [...this.state.AddedClasses, newClassObj.Id]
+      AddedClasses: [...this.state.AddedClasses, newClassObj.id]
     });
+    /*/ 
+      This is where the "Add Custom Class" API call exists 
+    /*/
+
+    // Use the split() function to split the string into an array based on the ' ' (space) delimiter
+    var classNameArray = newClassObj.id.split(' ');
+    // Store the first element of the array (i.e., "Summer") as lowercase into a variable
+    var classPrefix = classNameArray[0];
+    // Store the second element of the array (i.e., "1") into a variable and cast it to an integer
+    var classCode = parseInt(classNameArray[1]);
+    // Retreieves the integer from the Credit Amount and stores it as its own variable
+    var credits = parseInt(newClassObj.credits[0].split(" Credits"))
+
+    // Converts status (i.e. "Taken" or "Planned") into a boolean value
+    var courseStatus;
+    if (status === 'Taken') {
+      courseStatus = true;
+    } else {
+      courseStatus = false;
+    }
+
+    console.log(classPrefix)
+    console.log("\n")
+    console.log(classCode)
+    console.log("\n")
+    console.log(credits)
+    console.log("\n")
+    console.log(courseStatus)
+    console.log("\n")
+
+    let apiURL = "http://localhost:4001/course"
+    let jsonReq = {
+      "course_title": "Custom Class",
```

Comment:
Can you change this to be the value of the course name that the user inputs? Instead of hardcoding "Custom Class"